src/main/java/org/dependencytrack/event/kafka/processor/RepositoryMetaResultProcessor.java

```diff
@@ -105,28 +140,48 @@ private void synchronizeRepositoryMetaComponent(final PersistenceManager pm, fin
                 if (result.hasPublished()) {
                     persistentRepoMetaComponent.setPublished(new Date(result.getPublished().getSeconds() * 1000));
                 }
-                persistentRepoMetaComponent.setLastCheck(new Date(record.timestamp()));
-                pm.makePersistent(persistentRepoMetaComponent);
+                persistentRepoMetaComponent.setLastCheck(new Date(incomingAnalysisResultRecord.timestamp()));
+                return persistentRepoMetaComponent;
+            }
+        } else {
+            return null;
+        }
+    }
 
-                trx.commit();
-            } catch (JDODataStoreException e) {
-                // TODO: DataNucleus doesn't map constraint violation exceptions very well,
-                // so we have to depend on the exception of the underlying JDBC driver to
-                // tell us what happened. We currently only handle PostgreSQL, but we'll have
-                // to do the same for at least H2 and MSSQL.
-                if (ExceptionUtils.getRootCause(e) instanceof final SQLException se
-                        && PSQLState.UNIQUE_VIOLATION.getState().equals(se.getSQLState())) {
-                    continue; // Retry
-                }
+    private IntegrityMetaComponent synchronizeIntegrityMetaResult(final Record<String, AnalysisResult> incomingAnalysisResultRecord, QueryManager queryManager, PackageURL purl) {
+        final AnalysisResult result = incomingAnalysisResultRecord.value();
 
-                throw e;
-            } finally {
-                if (trx.isActive()) {
-                    trx.rollback();
-                }
-            }
+        IntegrityMetaComponent persistentIntegrityMetaComponent = queryManager.getIntegrityMetaComponent(purl.toString());
 
-            return;
+        if (persistentIntegrityMetaComponent.getStatus().equals(FetchStatus.PROCESSED)) {
+            LOGGER.warn("""
+                    Received hash information for %s that has already been processed; Discarding
+                    """.formatted(purl));
+            return null;
+        }
+        if (result.hasIntegrityMeta()) {
+            if (result.getIntegrityMeta().hasMd5() || result.getIntegrityMeta().hasSha1() || result.getIntegrityMeta().hasSha256()
+                    || result.getIntegrityMeta().hasSha512() || result.getIntegrityMeta().hasCurrentVersionPublished()) {
+                persistentIntegrityMetaComponent.setMd5(result.getIntegrityMeta().getMd5());
+                persistentIntegrityMetaComponent.setSha256(result.getIntegrityMeta().getSha256());
```

Comment:
Would result.getIntegrityMeta().getSha256() throw exception if integrityMeta objects md5 but doesn't have sha256 since if-condition passed in this case?

Reply:
done :)